packages/frontend/src/hooks/useCartesianChartConfig.ts

```diff
@@ -190,33 +190,37 @@ const useCartesianChartConfig = (
     useEffect(() => {
         const setAxes = (
             prev: Partial<Partial<CompleteCartesianChartLayout>> | undefined,
-            xField: string,
-            yField: string[],
+            xField: string | undefined,
+            yFields: string[],
         ) => {
-            const validYFields = prev?.yField
+            const isCurrentXFieldValid: boolean =
+                !!prev?.xField && availableFields.includes(prev.xField);
```

Comment:
don't we need to check here if `xField !== undefined` ?

Reply:
I don't think so. We should allow the xField to be changed to undefined if there aren't any available fields. 

E.g: User had xField and yFields defined. Then unselected the field used in xField. We need to set the xField to undefined.